### DISCUSSION TITLE
IP whitelist

### DISCUSSION THREAD

**Discussion by nehasjains** [t0bst4r/home-assistant-matter-hub#239] [Q&A] [answered]
Where are the IP whitelist settings ?

**Comment by t0bst4r** [comment, marked as answer]
You cannot configure it using the UI.

If you are using the native HA Addon, IPs are already blocked - only ingress is allowed. So nothing to do here. 

If you are using the docker image, you can configure it using an env variable `HAMH_HTTP_IP_WHITELIST` or with the config file if you use it. 
IMPORTANT: the env variable does only support a single entry, but you can use an ip range / cidr, too (e.g. `192.168.178.0/24` for all addresses between `192.168.178.0` and `192.168.178.255` - see [this page](https://www.ipaddressguide.com/cidr) for more info).

if you are using some custom deployment via `npm` you can (in addition to env and config file) also use the command line argument `--http-ip-whitelist`, and you can specify it multiple times.

See the [readme](https://github.com/t0bst4r/home-assistant-matter-hub?tab=readme-ov-file#2-configuration) for more details.

**Comment by nehasjains** [comment]
Got it. Thabk you so much.
